File: travel_expense_app/lib/features/auth/domain/entities/user_entity.dart
import 'package:equatable/equatable.dart';

class UserEntity extends Equatable {
  final String id;
  final String email;
  final String name;
  final String role; // e.g., 'Employee', 'Manager', 'Finance Executive', 'Compliance Officer', 'Admin'
  final String? department;
  final String? avatarUrl;

  const UserEntity({
    required this.id,
    required this.email,
    required this.name,
    required this.role,
    this.department,
    this.avatarUrl,
  });

  @override
  List<Object?> get props => [id, email, name, role, department, avatarUrl];
}

File: travel_expense_backend/src/modules/users/entities/user.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, ManyToOne, JoinColumn } from 'typeorm';

export enum UserRole {
  EMPLOYEE = 'Employee',
  MANAGER = 'Manager',
  FINANCE = 'Finance Executive',
  COMPLIANCE = 'Compliance Officer',
  ADMIN = 'Admin',
}

@Entity('users')
export class User {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ unique: true })
  email: string;

  @Column()
  name: string;

  @Column({
    type: process.env.OFFLINE_MODE === 'true' ? 'simple-enum' : 'enum',
    enum: UserRole,
    default: UserRole.EMPLOYEE,
  })
  role: UserRole;

  @Column({ nullable: true })
  department: string;

  @Column({ nullable: true })
  avatarUrl: string;

  @Column({ nullable: true })
  managerId: string;

  @ManyToOne(() => User, { nullable: true, onDelete: 'SET NULL' })
  @JoinColumn({ name: 'managerId' })
  manager: User;

  @CreateDateColumn()
  createdAt: Date;
}
